lib/WiFiNative/src/WifiClientPosix.inl

```diff
@@ -0,0 +1,241 @@
+ */
+static std::map<const WiFiClient*, struct pollfd> gConnections;
+
+/**
+ * The invalid socket value.
+ */
+static const int SOCK_INVAL = -1;
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+WiFiClient::~WiFiClient()
+{
+    stop(); /* Cleanup eventually open socket. */
+}
+
+uint8_t WiFiClient::connected() const
+{
+    bool isConnected = false;
+    auto iterSock    = gConnections.find(this);
+
+    if (gConnections.end() != iterSock)
+    {
+        isConnected = (iterSock->second.fd != SOCK_INVAL);
+    }
+
+    return isConnected;
+}
+
+uint8_t WiFiClient::connect(const IPAddress& addr, uint16_t port)
+{
+    uint8_t retval = 0U;
+    int     socketFd = SOCK_INVAL;
+
+    if (0U != connected())
+    {
+        /* Connect is called on an already connected client.
+         * Handle it as re-connect by closing the former socket connection.
+         */
+        stop();
+    }
+
+    socketFd = ::socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+    if (SOCK_INVAL != socketFd)
+    {
+        struct sockaddr_in serverAddr;
+
+        memset(&serverAddr, 0, sizeof(serverAddr));
+        serverAddr.sin_family      = AF_INET;
+        serverAddr.sin_addr.s_addr = htonl(addr);
+        serverAddr.sin_port        = htons(port);
+
+        if (0 == ::connect(socketFd, reinterpret_cast<struct sockaddr*>(&serverAddr), sizeof(serverAddr)))
+        {
+            const int enableOptVal = 1;
+
+            if (-1 == ::setsockopt(socketFd, IPPROTO_TCP, TCP_NODELAY, &enableOptVal, sizeof(enableOptVal)))
+            {
+                LOG_ERROR("%s:%s", "setsockopt", strerror(errno));
+            }
+            else if (-1 == ::fcntl(socketFd, F_SETFL, ::fcntl(socketFd, F_GETFL) | O_NONBLOCK))
+            {
+                LOG_ERROR("%s:%s", "fcntl", strerror(errno));
+            }
+            else
+            {
+                retval = 1U; /* success*/
+            }
+        }
+        else
+        {
+            LOG_ERROR("%s:%s", "connect", strerror(errno));
+        }
+
+        if (0U == retval)
+        {
+            stop(); /* One of the system calls above failed. */
+        }
+        else
+        {
+            /* Store poll data and listen to input data event. */
+            gConnections[this] = {.fd = socketFd, .events = POLLIN, .revents = 0};
+        }
+    }
+    else
+    {
+        LOG_ERROR("%s:%s", "socket", strerror(errno));
+    }
+
+    return retval;
+}
+
+void WiFiClient::stop()
+{
+    if (0U != connected())
+    {
+        auto iterSock = gConnections.find(this);
+
+        if (gConnections.end() != iterSock)
+        {
+            if (0 != ::close(iterSock->second.fd))
+            {
+                LOG_ERROR("%s:%s", "close", strerror(errno));
+            }
+            (void)Connections.erase(iterSock);
```

Comment:
Same iterator is used with different containers 'gConnections' and 'Connections'.

Reply:
typo :(